v2/slogger/rolling_file_appender/rolling_file_appender.go

```diff
@@ -18,59 +18,72 @@
 package rolling_file_appender
 
 import (
-	"fmt"
 	"github.com/mongodb/slogger/v2/slogger"
+
+	"fmt"
 	"os"
 	"path/filepath"
-	"regexp"
 	"sort"
-	"strconv"
 	"sync"
 	"time"
 )
 
 type RollingFileAppender struct {
-	MaxFileSize          int64
-	MaxRotatedLogs       int
+	maxFileSize          int64
+	maxDuration          time.Duration
+	maxRotatedLogs       int
 	file                 *os.File
 	absPath              string
 	curFileSize          int64
 	headerGenerator      func() []string
 	stringWriterCallback func(*os.File) slogger.StringWriter
 	lock                 sync.Mutex
+	state                *state
```

Comment:
Your understanding is correct.  I will add documentation but I will avoid adding comments that are likely to go stale, such as documenting that only the start timestamp is part of the state.  I could see documenting that the file size is not part of state but I think the struct itself is enough to document what's in it.